File: frontend/StudentManagement.Shared/Dtos/Admin/SystemHealthResponse.cs
namespace StudentManagement.Shared.Dtos.Admin;

public sealed class SystemHealthResponse
{
    public string DatabaseStatus { get; set; } = string.Empty;
    public string JvmVersion { get; set; } = string.Empty;
    public long TotalMemoryBytes { get; set; }
    public long FreeMemoryBytes { get; set; }
    public long UsedMemoryBytes { get; set; }
    public string OsName { get; set; } = string.Empty;
}

File: backend/src/main/java/com/demo/student_management/dto/admin/SystemHealthResponse.java
package com.demo.student_management.dto.admin;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class SystemHealthResponse {
    private String databaseStatus;
    private String jvmVersion;
    private long totalMemoryBytes;
    private long freeMemoryBytes;
    private long usedMemoryBytes;
    private String osName;
}
